test/widget_tests/sensor_widget_test.dart

```diff
@@ -0,0 +1,88 @@
+      await tester.pumpWidget(widget);
+
+      final dataTable = find.byType(DataTable);
+
+      expect(dataTable, findsOneWidget);
+    });
+
+    testWidgets('shows last refresh time', (tester) async {
+      final widget = TestableWidget(child: ViamSensorWidget(sensor: FakeSensor('sensor'), showLastRefreshed: true));
+      await tester.pumpWidget(widget);
+
+      final refreshButton = find.widgetWithIcon(ViamButton, Icons.refresh);
+      await tester.press(refreshButton);
+      await tester.pump();
+
+      final refreshTime = find.textContaining(RegExp(r'Updated at:.*'));
+
+      expect(refreshTime, findsOneWidget);
+    });
+
+    testWidgets('hides last refresh time', (tester) async {
+      final widget = TestableWidget(child: ViamSensorWidget(sensor: FakeSensor('sensor'), showLastRefreshed: false));
+      await tester.pumpWidget(widget);
+
+      final refreshButton = find.widgetWithIcon(ViamButton, Icons.refresh);
+      await tester.press(refreshButton);
+      await tester.pump();
+
+      final refreshTime = find.textContaining(RegExp(r'Updated at:.*'));
+
+      expect(refreshTime, findsNothing);
+    });
+
+    testWidgets('shows refresh controls', (tester) async {
+      final widget = TestableWidget(child: ViamSensorWidget(sensor: FakeSensor('sensor'), showRefreshControls: true));
+      await tester.pumpWidget(widget);
+
+      final playButton = find.widgetWithIcon(ViamButton, Icons.play_arrow);
+      final pauseButton = find.widgetWithIcon(ViamButton, Icons.pause);
+      final refreshButton = find.widgetWithIcon(ViamButton, Icons.refresh);
+
+      expect(playButton, findsNothing);
+      expect(pauseButton, findsOneWidget);
+      expect(refreshButton, findsOneWidget);
+
+      await tester.press(pauseButton);
+      await tester.pump();
+
+      expect(playButton, findsNothing);
+      expect(pauseButton, findsOneWidget);
+    });
+
+    testWidgets('hides refresh controls', (tester) async {
+      final widget = TestableWidget(child: ViamSensorWidget(sensor: FakeSensor('sensor'), showRefreshControls: false));
+      await tester.pumpWidget(widget);
+
+      final playButton = find.widgetWithIcon(ViamButton, Icons.play_arrow);
+      final pauseButton = find.widgetWithIcon(ViamButton, Icons.pause);
+      final refreshButton = find.widgetWithIcon(ViamButton, Icons.refresh);
+
+      expect(playButton, findsNothing);
+      expect(pauseButton, findsNothing);
+      expect(refreshButton, findsNothing);
+    });
+
+    testWidgets('hides play/pause when refresh interval is null', (tester) async {
+      final widget = TestableWidget(child: ViamSensorWidget(sensor: FakeSensor('sensor'), refreshInterval: null));
+      await tester.pumpWidget(widget);
+
+      final playButton = find.widgetWithIcon(ViamButton, Icons.play_arrow);
+      final pauseButton = find.widgetWithIcon(ViamButton, Icons.pause);
+
+      expect(playButton, findsNothing);
+      expect(pauseButton, findsNothing);
+    });
+  });
```

Comment:
(q) maybe obvious but does it also hide the refresh button? should we still test for that?

Reply:
Nope! When `refreshInterval` is `null`, only the play/pause button is hidden. The manual refresh button **needs** to be available so that users can manually refresh

Reply:
But you are correct that we should have a test to make sure the refresh button **is** visible so I added it :)